Composer/packages/ui-plugins/select-skill-dialog/src/BeginSkillDialogField.tsx

```diff
@@ -1,66 +1,132 @@
+    const skill = skills.find(({ name }) => name === selectedSkill);
+    return skill;
+  }, [skills, selectedSkill]);
 
-  const handleEndpointChange = (skillEndpoint) => {
-    const { msAppId } =
-      (manifest?.endpoints || []).find(({ endpointUrl }) => endpointUrl === skillEndpoint) || ({} as any);
-    onChange({ ...value, skillEndpoint, ...(msAppId ? { skillAppId: msAppId } : {}) });
+  const endpointOptions = useMemo(() => {
+    return (matchedSkill?.endpoints || []).map(({ name }) => name);
+  }, [matchedSkill]);
+
+  const handleEndpointChange = async (skillEndpoint) => {
+    if (matchedSkill) {
+      const { msAppId, endpointUrl } =
+        (matchedSkill.endpoints || []).find(({ name }) => name === skillEndpoint) || ({} as any);
+      const schemaUpdate: any = {};
+      const settingsUpdate: any = {};
+      if (endpointUrl) {
+        skillsInSettings.set(matchedSkill.name, { endpointUrl });
+        schemaUpdate.skillEndpoint = referBySettings(matchedSkill?.name, 'endpointUrl');
+        settingsUpdate.endpointUrl = endpointUrl;
+      }
+      if (msAppId) {
+        schemaUpdate.skillAppId = referBySettings(matchedSkill?.name, 'msAppId');
+        settingsUpdate.msAppId = msAppId;
+      }
+      skillsInSettings.set(matchedSkill.name, { ...settingsUpdate });
+      onChange({
+        ...value,
+        ...schemaUpdate,
+      });
+    }
   };
 
+  useEffect(() => {
+    if (oldEndpoint) {
+      handleEndpointChange(oldEndpoint);
+    }
+  }, [oldEndpoint]);
+
   const handleShowManifestClick = () => {
-    value.id && displayManifestModal(value.id);
+    matchedSkill && displayManifestModal(matchedSkill.manifestUrl);
   };
 
   const skillEndpointUiSchema = uiOptions.properties?.skillEndpoint || {};
   skillEndpointUiSchema.serializer = {
     get: (value) => {
-      const endpoint = (manifest?.endpoints || []).find(({ endpointUrl }) => endpointUrl === value);
+      const url: any = skillsInSettings.get(value);
+      const endpoint = (matchedSkill?.endpoints || []).find(({ endpointUrl }) => endpointUrl === url);
       return endpoint?.name;
     },
     set: (value) => {
-      const endpoint = (manifest?.endpoints || []).find(({ name }) => name === value);
+      const endpoint = (matchedSkill?.endpoints || []).find(({ name }) => name === value);
       return endpoint?.endpointUrl;
     },
   };
 
+  const onSkillSelectionChange = (option: IComboBoxOption | null) => {
+    if (option) {
+      setSelectedSkill(option?.text);
+    }
+  };
+
   return (
     <React.Fragment>
-      <SchemaField
-        definitions={definitions}
-        depth={depth + 1}
-        id={`${id}.id`}
-        name="id"
-        rawErrors={{}}
-        schema={(schema?.properties?.id as JSONSchema7) || {}}
-        uiOptions={uiOptions.properties?.id || {}}
-        value={value?.id}
-        onChange={handleIdChange}
-      />
+      <SelectSkillDialog depth={depth + 1} value={selectedSkill} onChange={onSkillSelectionChange} />
```

Comment:
I don't think we need to pass `depth` as a prop here. We can just use a value greater than 0 in the component.